llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -2305,16 +2305,27 @@ Value *InnerLoopVectorizer::createIterationCountCheck(ElementCount VF,
   Type *CountTy = Count->getType();
   Value *CheckMinIters = Builder.getFalse();
   auto CreateStep = [&]() -> Value * {
+    ElementCount VFTimesUF = VF.multiplyCoefficientBy(UF);
+    Value *VFxUF = nullptr;
+    if (!VFTimesUF.isScalable() ||
+        !isPowerOf2_64(VFTimesUF.getKnownMinValue())) {
+      VFxUF = createStepForVF(Builder, CountTy, VF, UF);
+    } else {
+      VFxUF = Builder.CreateShl(
+          Builder.CreateVScale(CountTy),
+          ConstantInt::get(CountTy, Log2_64(VFTimesUF.getKnownMinValue())), "",
+          true);
+    }
+
```

Comment:
Can we move this into createStepForVF? `isPowerOf2_64(VFTimesUF.getKnownMinValue())` should be equivalent to `isPowerOf2_64(VF.getKnownMinValue()) && isPowerOf2_64(Step)`

Reply:
I originally thought that this probably adds a lot to the diff, but it doesn't seem to bad, so I added this to the patch here, thanks